src/x/serialize/decoder_fast_iter.go

```diff
@@ -0,0 +1,85 @@
+package serialize
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/m3db/m3/src/metrics/metric/id"
+)
+
+type decoderFast struct {
+	length    int
+	remaining int
+
+	encodedTags []byte
+
+	currentName, currentValue []byte
+
+	err error
+}
+
+func NewFastTagDecoder() id.SortedTagIterator {
```

Comment:
The comment of `SortedTagIterator`:  `iterates over a set of tag pairs sorted by tag names.`
This decoder returns tags in the same order they appear in `encodedTags`, can we be sure we can provide this guarantee?